### PR TITLE
Add docs for 'ISessionStore',  'DistributedSessionStore' and 'DistributedSession'

### PR DESCRIPTION
**Pull request by hishamco**

### CODE REVIEW

File: src/Microsoft.AspNet.Session/DistributedSession.cs [hunk 1]
```diff
@@ -14,6 +14,9 @@
 
 namespace Microsoft.AspNet.Session
 {
+    /// <summary>
+    /// Represent a distributed cache session.".
```
Comment (Eilon) [inline src/Microsoft.AspNet.Session/DistributedSession.cs]: Extra quote/period.

Comment (Eilon) [inline src/Microsoft.AspNet.Session/DistributedSession.cs]: Plus I'm not sure this comment really says much about what it is. It's just kind of saying the type name.

File: src/Microsoft.AspNet.Session/DistributedSession.cs [hunk 2]
```diff
@@ -29,6 +32,15 @@ public class DistributedSession : ISession
         private bool _loaded;
         private bool _isNewSessionKey;
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DistributedSession"/> class.
+        /// </summary>
+        /// <param name="cache">The <see cref="IDistributedCache"/>.</param>
+        /// <param name="sessionId">Represents the session id.</param>
+        /// <param name="idleTimeout">Represents the session timeout.</param>
+        /// <param name="tryEstablishSession">Represents the delegate that trying to establish the session.</param>
```
Comment (Eilon) [inline src/Microsoft.AspNet.Session/DistributedSession.cs]: Grammar

File: src/Microsoft.AspNet.Session/DistributedSession.cs [hunk 3]
```diff
@@ -29,6 +32,15 @@ public class DistributedSession : ISession
...
+        /// <param name="loggerFactory">The <see cref="ILoggerFactory"/>.</param>
+        /// <param name="isNewSessionKey">Determine whether the session key is new.</param>
```
Comment (Eilon) [inline src/Microsoft.AspNet.Session/DistributedSession.cs]: What does this actually mean?

File: src/Microsoft.AspNet.Session/DistributedSessionStore.cs [hunk 4]
```diff
@@ -9,17 +9,26 @@
 
 namespace Microsoft.AspNet.Session
 {
+    /// <summary>
+    /// Represent a distributed cache session store.".
```
Comment (Eilon) [inline src/Microsoft.AspNet.Session/DistributedSessionStore.cs]: See earlier comment

File: src/Microsoft.AspNet.Session/ISessionStore.cs [hunk 5]
```diff
@@ -6,10 +6,29 @@
 
 namespace Microsoft.AspNet.Session
 {
+    /// <summary>
+    /// Represent a session store.".
```
Comment (Eilon) [inline src/Microsoft.AspNet.Session/ISessionStore.cs]: All these have extra quotes/periods

File: src/Microsoft.AspNet.Session/ISessionStore.cs [hunk 6]
```diff
@@ -6,10 +6,29 @@
...
+        /// </summary>
         void Connect();
+        
+        /// <summary>
+        /// Create <see cref="ISession"/>.
+        /// </summary>
+        /// <param name="sessionId">Represents the session id.</param>
+        /// <param name="idleTimeout">Represents the session timeout.</param>
+        /// <param name="tryEstablishSession">Represents the delegate that trying to establish the session.</param>
+        /// <param name="isNewSessionKey">Determine whether the session key is new.</param>
```
Comment (Eilon) [inline src/Microsoft.AspNet.Session/ISessionStore.cs]: See earlier comments

File: src/Microsoft.AspNet.Session/ISessionStore.cs [hunk 7]
```diff
@@ -6,10 +6,29 @@
 
 namespace Microsoft.AspNet.Session
 {
+    /// <summary>
+    /// Represent a session store.".
+    /// </summary>
     public interface ISessionStore
     {
+        /// <summary>
+        /// Determine the session availability.".
```
Comment (Eilon) [inline src/Microsoft.AspNet.Session/ISessionStore.cs]: What does this actually mean?

**Comment by Eilon** [comment]
I urge new doc comments to really help a user understand what these types are, what the methods do, and what the parameters represent. The doc comments here might be a good start, but they do little to add value beyond just the type names and member names.

**Comment by hishamco** [comment]
Sadly i'm not sure where the extra quote is come from :( and thanks for your feedback @Eilon, I will try to improve it as much as I can

**Comment by muratg** [comment]
@Eilon let's discuss what approach we will take for doc comments. 

@glennc @DamianEdwards

**Comment by muratg** [comment]
Closing this PR for now. We'll discuss the doc comments (style, what information to present etc.) to have a coherent set of API docs for an upcoming milestone.

**Comment by muratg** [comment]
Thanks @hishamco!

**Pull request closed without merging**